Bluetooth: controller: Refactor sync info population

cvinayak opened the pull request:
Refactor out sync info field population into a utility
function with set and clear interface to add or remove
the common extended advertising header format fields.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Refactor sync info population

Refactor out sync info field population into a utility
function with set and clear interface to add or remove
the common extended advertising header format fields.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
tests: Bluetooth: bsim_test: tests periodic adv enable/disable

Update bsim test to test Periodic Advertising Enable and
Disable alongwith Extended Advertising Enable/Disable.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

kruithofa commented on tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:
```diff
@@ -145,15 +144,18 @@ static void test_advx_main(void)
 	if (err) {
 		goto exit;
 	}
+	printk("success.\n");
```
we should verify that advertising indeed has started i.e. sending packets over the air. We can then also remove the following sleep

cvinayak replied:
Is there a bsim API to synchronize out of band between devices?

kruithofa replied:
Good question, a quick search didn't show one

wopu-ot replied:
Checking air traffic can be done by having the test script check the logs after the fact. Alternatively, EDTT coordinates two or more devices and can check whether for example advertisements from one device are seen by another device. Maybe something like that would be the way to go here?

kruithofa commented on subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:
```diff
@@ -589,6 +589,395 @@ int ull_adv_aux_reset(void)
 	return 0;
 }
 
+uint8_t ull_adv_aux_hdr_set_clear(struct ll_adv_set *adv,
```
This function is still quite large, and IMO should be split up even further

cvinayak replied:
Suggestions?

kruithofa replied:
I would think good candidates for a function would include (note I'm not 100% sure about naming convention we should use for these static inline functions):
lines 613-627: adv_acquire_lll_aux
lines 630-645: adv_fill_pri_pdu
lines 648-678: adv_fill_sec_pdu
lines 775-782: adv_calc_pri_pdu_len
lines 784-812: adv_calc_sec_pdu_len
lines 844-868: adv_fill__sec-pdu_sync
lines 872-892: adv_fill_aux_pri_pdu
lines 895-911: adv_fill_aux_sec_pdu

andrzej-kaczmarek replied:
perhaps this could be also called smth like `ull_adv_aux_configure` - the same code can be used to basically create an aux for advertising instance without adding or removing sync info. it would be used in e.g. `ll_adv_params_set` when configured instance needs aux (scannable or connectable) or later in `ll_adv_aux_ad_data_set` where aux is now created by pretty much the same code.

cvinayak replied:
Code blocks that was repeated across ull_adv, ull_adv_aux, and ull_adv_sync have been refactored to be reused. The function has not been split up further due to the fact that this function needs to be refactored based on how scan response and chaining is to be handled. 

Any further refactoring/split is viable only after(/ are part of) scannable mode support.

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:
```diff
@@ -577,7 +231,31 @@ uint8_t ll_adv_sync_enable(uint8_t handle, uint8_t enable)
 	if (adv->is_enabled && !sync->is_started) {
-		/* TODO: */
+		volatile uint32_t ret_cb = TICKER_STATUS_BUSY;
+		uint32_t ticks_anc_sync;
```
agree.

cvinayak committed:
Bluetooth: controller: Refactor out AD data population

Refactor out AD data population to use the common function
ull_adv_aux_hdr_set_clear.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Refactor to allow code reuse

Moved the assignment of PDU length out side of a conditional
so that code to calculate PDU length can be reused.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Add aux ptr fill function

Refactor out into function the filling of aux ptr structure
in the common extended advertising payload format header.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Add sync info structure fill function

Refactor out into function the filling of sync info
structure in the common extended advertising payload format
header.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Add common header len get/fill function

Refactor out into functions the calculation of common
extended advertising payload format header length.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

kruithofa commented on tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c:
```diff
@@ -436,6 +441,40 @@ static void test_advx_main(void)
 	}
 	printk("success.\n");
 
+	k_sleep(K_MSEC(1000));
```
I needed to increase the sim_length in the basic_advx.sh script to 5e7 for these tests to pass on my system. Please increase this value

cvinayak committed:
Bluetooth: controller: Review rework use HCI defines

Use HCI error and command parameter defines.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
tests: Bluetooth: bsim_advx_test: Increase sim_length to 60 seconds

Increased sim_length value to 60 seconds so as to meet the
current length of 13.56 seconds and any future test timing
increases in the near future.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

andrzej-kaczmarek commented on subsys/bluetooth/controller/ll_sw/ull_adv_aux.c:
```diff
@@ -120,348 +111,16 @@ uint8_t ll_adv_aux_ad_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref, ui
-	pri_pdu_prev = lll_adv_data_peek(lll);
-	if (pri_pdu_prev->type != PDU_ADV_TYPE_EXT_IND) {
-		return ull_adv_data_set(adv, len, data);
-	}
```
I may be missing something, but how is setting data on a legacy instance handled now?

cvinayak commented:
@kruithofa I run the bsim tests as below:

```
# cd build
# ./zephyr/zephyr.exe -s=advx -d=0 -testid=advx
```
```
# cd build
# ./zephyr/zephyr.exe -s=advx -d=1 -testid=scanx
```
```
# cd ~/workspace/bsim/bin
# bs_2G4_phy_v1 -s=advx -D=2 -sim_length=20e6
```

or

```
# cd $ZEPHYR_BASE
# bash ./tests/bluetooth/bsim_bt/bsim_test_advx/tests_scripts/basic_advx.sh
```

kruithofa reviewed (approved):
LGTM

andrzej-kaczmarek reviewed (approved):
lgtm
